MAUI/DataForm/Data-annotations.md

```diff
@@ -0,0 +1,374 @@
+Specifies watermark text for the editor.
+</td>
+</tr>
+<tr>
+<td>
+{{'[Order](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute.order?view=net-7.0)' | markdownify }}
+</td>
+<td>
+Specifies the order of field in the data form.
+</td>
+</tr>
+</table>
+
+{% tabs %}
+{% highlight C# }
+
+    [Display(Name = "First Name", GroupName = "Name", Prompt="Enter your name")]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Validation attribute
+
+<table>
+<tr>
+<th>
+Property
+</th>
+<th>
+Details
+</th>
+</tr>
+<tr>
+<td>
+{{'[MinLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.minlengthattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required minimum input string length.
+</td>
+</tr>
+<tr>
+<td>
+{{'[MaxLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.maxlengthattribute?view=net-7.0)'| markdownify }}
+
+</td>
+<td>
+Specifies the required maximum input string length.
+</td>
+</tr>
+<tr>
+<td>
+{{'[Required](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.requiredattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies that the data field value is required. 
+</td>
+</tr>
+<tr>
+<td>
+{{'[StringLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.stringlengthattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required string length. 
+</td>
+</tr>
+<tr>
+<td>
+{{'[EnumDataType](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.enumdatatypeattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies enum type for the data field.
+</td>
+</tr>
+</table>
+
+{% tabs %}
+{% highlight MinLength attribute}
+
+    [MinLength(5,ErrorMessage = "Password length must be greater than 5 characters")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight MaxLength attribute}
+
+    [MaxLength(20,ErrorMessage = "Maximum password length should be less than 20")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Required attribute}
+
+    [Required(AllowEmptyStrings = false, ErrorMessage ="Name is required")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight StringLength attribute}
+
+    [StringLength(20,ErrorMessage = "Name should not exceed 20 characters")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+
+{% tabs %}
+{% highlight EnumDataType attribute}
+
+    [EnumDataType(typeof(Gender), ErrorMessage = "Please select Gender")]
+    public Gender Gender { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Bindable attribute
+
+It specifies whether the field should be auto generated or not. If bindable set to false the field will not be auto generated. 
```

Comment:
It specifies whether the field should be auto-generated or not. If bindable is set to false, the field will not be auto-generated.

Reply:
- added